resolver/src/lookup_state.rs

```diff
@@ -239,61 +240,54 @@ impl<C: DnsHandle<Error = ResolveError> + 'static> QueryFuture<C> {
 
     fn handle_noerror(&mut self, mut message: Message) -> Poll<Records, ResolveError> {
         // seek out CNAMES
-        // TODO: figure out how to get rid of this clone
-        let mut cname_ttl = 0;
-        let mut was_cname = false;
-        let mut search_name: Name = self.query.name().clone();
-        while let Some(cname) = message.answers().iter().find(|r| {
-            r.rr_type() == RecordType::CNAME && r.name() == &search_name
-        })
-        {
-            was_cname = true;
-            cname_ttl = cname.ttl();
-            if let &RData::CNAME(ref name) = cname.rdata() {
-                if search_name == *name {
-                    break; // already searched for this name
-                } else {
-                    search_name = name.clone();
-                }
-            } else {
-                // now that is very odd...
-                warn!("Expected RData::CNAME in response record {:?}", cname);
```

Comment:
I don't understand how we could encounter a non-CNAME RData in a record that has a record type of CName. Won't the parser guarantee that these two fields are consistent?

Reply:
this is just me handling a potential error condition. You are correct that it should never happen, an assert would have been as valid...